lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -407,60 +566,160 @@ void _AddMissingTexcoordReaders(mx::DocumentPtr& mtlxDoc)
                     node->getNamePath().c_str())) {
                 return;
             }
-            for (mx::InputPtr input : nodeDef->getInputs()) {
-                if (input->hasDefaultGeomPropString()
-                    && input->getDefaultGeomPropString() == _mtlxTokens->UV0.GetString()) {
-                    // See if the corresponding input is connected on the node:
-                    if (node->getConnectedNodeName(input->getName()).empty()) {
-                        // Create emergency ST reader if necessary
-                        if (!stReader) {
-                            stReader = nodeGraph->addNode(
-                                _mtlxTokens->geompropvalue.GetString(),
-                                _mtlxTokens->ST_reader.GetString(),
-                                _mtlxTokens->vector2.GetString());
-                            mx::ValueElementPtr prpInput
-                                = stReader->addInputFromNodeDef(_mtlxTokens->geomprop.GetString());
-                            prpInput->setValueString(_tokens->st.GetString());
-                        }
-                        node->addInputFromNodeDef(input->getName());
-                        node->setConnectedNodeName(input->getName(), stReader->getName());
-                    }
+
+            if (!stReader && nodeDef->getNodeString() == _mtlxTokens->geompropvalue.GetString()
+                && node->getType() == _mtlxTokens->vector2.GetString()) {
+                // Grab the first st reader we can find. This will be the default one used for
+                // tangents unless we find something better.
+                stReader = node;
+                continue;
```

Comment:
I'm not sure I understand why we can continue here. The first node def that matches will get the continue, the later if another nodeDef matches we execute the rest of the loop and maybe have a different behavior. I don't understand it in depth enough to know if it is a problem or not, it just looks a bit strange.

Reply:
We can continue here because we know that a geompropvalue node never has a tangent input. Note that we do not continue in the next branch because normalmap nodes have tangent input.